src/Utils/Req.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Utils.Req (
+  showHTTPException
+  ) where
+
+import Data.ByteString.Char8 (ByteString)
+import Data.Text (Text, pack)
+import qualified Network.HTTP.Client as Client
+import qualified Network.HTTP.Req as Req
+import Network.HTTP.Types (Status(..))
+
+import PyF (fmt)
+
+showHTTPException :: Req.HttpException
+                  -> Text
+showHTTPException (Req.VanillaHttpException clientHttpException) = showClientHttpException clientHttpException
+showHTTPException (Req.JsonHttpException exc) = pack exc
+
+showClientHttpException :: Client.HttpException
+                        -> Text
+showClientHttpException (Client.HttpExceptionRequest _ excContent) = showExceptionContent excContent
+showClientHttpException (Client.InvalidUrlException _ reason) = pack reason
+
+showExceptionContent :: Client.HttpExceptionContent
+                     -> Text
+showExceptionContent (Client.StatusCodeException resp body) = showResponse resp body
+-- Catch all doing a simple and ugly show
+showExceptionContent exc = pack . show $ exc
+
+showResponse :: Client.Response ()
+             -> ByteString
+             -> Text
+showResponse resp body = [fmt|\
+HTTP call failed: {show status} - {show statusMsg}
+      {show body}\
```

Comment:
I don't like that. The user of the `Text` returned actually displays it indented by 6 spaces. so if I return a multiline string, I need to prefix all lines, except the first one, with 6 spaces. That means I need to know details of my caller. this is not good. We should find a better way.

For example the consumer of this method could indent each line of the `Text` returned, but I guess this is rather inefficient.

Ideas?

Reply:
I do agree that it is not beautiful. However I think it is orthogonal to this pull request. The impact factor of your change is already high because it gives context to errors and I'm sure that it will help user, as in #49. So I see no reasons to report the merge. We'll fix formatting later.